File: java/src/leetcode/easy/LinkedListCycle.java
package leetcode.easy;

import common.ListNode;

/**
 * Question: https://leetcode.com/problems/linked-list-cycle/description/
 */
public class LinkedListCycle {
    public boolean hasCycle(ListNode head) {
        ListNode current = head, fast = head;
        while (fast != null && fast.next != null) {
            current = current.next;
            fast = fast.next.next;
            if (current.equals(fast)) {
                return true;
            }
        }
        return false;
    }
}

File: javascript/easy/linkedListCycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let current = head, runner = head;
    while(runner != null && runner.next != null) {
        current = current.next;
        runner = runner.next.next;
        if(current == runner) {
            return true;
        }
    }
    return false;
};
